### PR TITLE
BRS-746: Added ability to move passes out of overbooked.

### PR DESCRIPTION
**Pull request by danieltruong**
### Jira Ticket:

BRS-746

### Jira Ticket URL:

https://bcparksdigital.atlassian.net/browse/BRS-746

### Description:

Added ability to move passes out of overbooked.

### CODE REVIEW

File: lambda/writeFacility/index.js [hunk 1]
```diff
@@ -289,6 +305,71 @@ async function updateReservationsObjectCapacity(pk, sk, type, newBaseCapacity, n
...
+    IndexName: 'shortPassDate-index',
+    ExpressionAttributeValues: {
+      ':shortPassDate': { S: shortPassDate },
+      ':facilityName': { S: facilityName },
+      ':passType': { S: type },
+      ':true': { BOOL: true }
+    },
+    ExpressionAttributeNames: {
+      '#theType': 'type',
+      '#isOverbooked': 'isOverbooked'
```
Comment (marklise) [inline lambda/writeFacility/index.js:320]: Not a reserved word, so no need to include in ExpressionAttributeNames

File: lambda/writeFacility/index.js [hunk 2]
```diff
@@ -252,8 +252,24 @@ async function processReservationObjects(resObjs, timesToUpdate) {
...
+        let overbookedPasses = [];
+        try {
+          overbookedPasses = await checkForOverbookedPasses(resObj.pk.split('::').pop(), resObj.sk, timeToUpdate.time);
+        } catch (error) {
+          logger.error('Error occured while executing checkForOverbookedPasses()');
+          throw error;
+        }
+        if (overbookedPasses.length > 0) {
+          try {
+            newResAvailability = await reverseOverbookedPasses(overbookedPasses, passDiff);
```
Comment (cameronpettit) [inline lambda/writeFacility/index.js:266]: Change to
```
newResAvailability -= await reverseOverbookedPasses(overbookedPasses, newResAvailability);
```

File: lambda/writeFacility/index.js [hunk 3]
```diff
@@ -289,6 +305,71 @@ async function updateReservationsObjectCapacity(pk, sk, type, newBaseCapacity, n
...
+    try {
+      const res = await dynamodb.updateItem(updatePassObject).promise();
+      logger.debug('Reversed pass overbooked status', res);
+    } catch (error) {
+      logger.error('Error occured while updating pass in reverseOverbookedPasses');
+      logger.error(updatePassObject);
+      throw { msg: 'Something went wrong.', title: 'Operation Failed' };
+    }
+  }
+  return passDiff - passTally;
```
Comment (cameronpettit) [inline lambda/writeFacility/index.js:375]: Change to 
```
return passTally;
```

**Review by cameronpettit — CHANGES_REQUESTED** [review]
Math is still not cooperating. Made these changes and it works fully on my end now.

File: lambda/writeFacility/index.js (hunk 2 above)
Comment (cameronpettit) [reply lambda/writeFacility/index.js:266]: Better yet, separate it out:
```
const restoredPassQuantity = await reverseOverbookedPasses(overbookedPasses, newResAvailability);
newResAvailability -= restoredPassQuantity;
```

File: lambda/writeFacility/index.js [hunk 4]
```diff
@@ -289,6 +305,71 @@ async function updateReservationsObjectCapacity(pk, sk, type, newBaseCapacity, n
...
+      ':facilityName': { S: facilityName },
+      ':passType': { S: type },
+      ':true': { BOOL: true }
+    },
+    ExpressionAttributeNames: {
+      '#theType': 'type',
+      '#isOverbooked': 'isOverbooked'
+    },
+    KeyConditionExpression: 'shortPassDate =:shortPassDate AND facilityName =:facilityName',
+    FilterExpression: '#theType =:passType AND #isOverbooked =:true'
```
Comment (cameronpettit) [inline lambda/writeFacility/index.js]: I think we will also have to filter out `cancelled` passes here.

**Pull request closed without merging**